### PR TITLE
Make repo mirror URLs configurable via environmental variable

### PR DESCRIPTION
**Pull request by donat**
Currently, the internal repository mirrors are hardcoded in the build script. This means every time we want to change repository mirrors we need to adjust the build script. 

This change makes the mirror URLs read from an environmental variable and passes the values to the integration tests. The env variable is already set up in TeamCity.

Also, I think we should merge this PR both to the master/release branch.

### CODE REVIEW

File: gradle/distributionTesting.gradle [hunk 1]
```diff
@@ -46,11 +46,12 @@ distributionTestTasks.all { DistributionTest task ->
 
     systemProperties['org.gradle.integtest.multiversion'] = project.hasProperty("testAllVersions") && project.testAllVersions ? 'all' : 'default'
 
-    systemProperties['org.gradle.integtest.mirrors.mavencentral'] = isCiServer ? 'http://dev12.gradle.org:8081/artifactory/repo1' : ''
-    systemProperties['org.gradle.integtest.mirrors.jcenter'] = isCiServer ? 'http://dev12.gradle.org:8081/artifactory/jcenter' : ''
-    systemProperties['org.gradle.integtest.mirrors.typesafemaven'] = isCiServer ? 'http://dev12.gradle.org:8081/artifactory/typesafe-maven-releases' : ''
-    systemProperties['org.gradle.integtest.mirrors.typesafeivy'] = isCiServer ? 'http://dev12.gradle.org:8081/artifactory/typesafe-ivy-releases' : ''
-    systemProperties['org.gradle.integtest.mirrors.google'] = isCiServer ? 'http://dev12.gradle.org:8081/artifactory/google' : ''
+    def mirrorUrls = collectMirrorUrls()
+    systemProperties['org.gradle.integtest.mirrors.mavencentral'] = mirrorUrls['mavencentral'] ?: ''
```
Comment (bamboo) [inline gradle/distributionTesting.gradle]: :x: there's a lot of duplication here, I would keep the names of the expected mirrors in a list and loop, something like:
```Groovy
def mirrors = ['mavencentral', 'jcenter', 'typesafemaven', 'typesafeivy', 'google']
mirrors.each { mirror ->
    systemProperties["org.gradle.integtest.mirrors.$mirror"] = mirrorUrls[mirror] ?: ''
}
```

File: gradle/distributionTesting.gradle [hunk 2]
```diff
@@ -106,6 +107,15 @@ distributionTestTasks.all { DistributionTest task ->
     }
 }
 
+def collectMirrorUrls() {
+    // expected env var format: repo1_id:repo1_url,repo2_id:repo2_url,...
+    def mirrors = System.getenv().get('REPO_MIRROR_URLS')
+    !mirrors ? [:] : mirrors.split(',').collectEntries { nameToUrl ->
```
Comment (bamboo) [inline gradle/distributionTesting.gradle]: ⭕️  It took me a little longer to decode the conditional logic in `!mirrors ? [:] : mirrors.split...`, not a huge deal but I think null-safe navigation followed by null coalescing might be easier to parse, I would even ditch the local variable so the whole method becomes a single expression:

```
System.getenv('REPO_MIRROR_URLS')?.split(',')?.collectEntries { nameToUrl ->
    ...
} ?: [:]
```

Comment (donat) [reply gradle/distributionTesting.gradle]: Yes, that's much more readable. I've fixed that.

File: gradle/distributionTesting.gradle (hunk 1 above)
Comment (donat) [reply gradle/distributionTesting.gradle]: Fixed.

**Pull request merged**